src/ui/organisms/popup.js

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useRef } from "react"
+import PropTypes from "prop-types"
+
+export const PopUp = ({ children, onClose }) => {
+  const ref = useRef(null)
+  const onDocumentClick = (event) => {
```

Comment:
Please use `useCallback` here.

because in `useEffect` on unmount used new function, but new function do not removed